### ISSUE TITLE
Horizontal scrollbar between 1220px and 1234px

### ISSUE THREAD

**Issue by makshh** [squidfunk/mkdocs-material#542]
## Description

There is a horizontal scrollbar between 1220px and 1234px screen width when you are at the top of the page (it disappears when sidebar on the right becomes fixed)

### Expected behavior

No horizontal scrollbar

### Actual behavior

Horizontal scrollbar between 1220px and 1234px screen width when you are at the top of the page

### Package versions

* MkDocs: 0.16.3
* Material: 1.12.1

### System information

* OS: OSX
* Browser: Chrome/Firefox

![](https://image.prntscr.com/image/fDvUF0eYS0Wxpe2TiJQkQA.png)

Possible fix:

```css
body {
  position: relative; // ?
  overflow-x: hidden; // ?
}

**Comment by squidfunk** [comment]
Hmm, it doesn't appear on Chrome or Firefox on OS X for the hosted docs. Did you change any CSS?

**Comment by makshh** [comment]
No, the screenshot comes from hosted docs. Strange, I also checked on Chrome/Windows and the same thing. Make sure you are at the top of the page (when right sidebar is not fixed).

[Horizontal scrollbar](https://media.giphy.com/media/3ohhwpI67kqZTlZl3G/giphy.gif)

**Comment by squidfunk** [comment]
I'm sorry, it's not reproducible for me. Someone else experiencing this? Please provide some more information on this.

**Comment by makshh** [comment]
Here is another screenshot (Windows 7/Chrome 61.0.3163.100)

![123124](https://user-images.githubusercontent.com/5025950/32135036-845c7282-bbf8-11e7-97ec-c281c870d071.png)

**Comment by squidfunk** [comment]
Is this http://squidfunk.github.io//mkdocs-material? Do you hve any browser extensions installed?

**Comment by makshh** [comment]
Yes, it is http://squidfunk.github.io/mkdocs-material/

And yes, I have some browser extensions, but I checked it also on Firefox without extensions and on Windows in Chrome without extensions in incognito mode.

I also asked my friend to go to the material docs, change browser width to 1225px, be at the top of the page and tell me if you see horizontal scrollbar, the answer was yes ;)

**Comment by mtik00** [comment]
Here's a screen grab on Windows 10 with Chrome Version 61.0.3163.100 (Official Build) (64-bit).  Same thing @makshh is seeing.  I've disabled the only browser extension that was installed (UBlock Origin).

![horizontal-bar](https://user-images.githubusercontent.com/1509355/32149052-d2ec62dc-bcc4-11e7-9847-d310209f670d.gif)

**Comment by squidfunk** [comment]
I can finally reproduce it on Windows, but not on OS X. Very strange. Will look into that.

**Comment by squidfunk** [comment]
So the problem is exactly this:
http://www.456bereastreet.com/archive/201301/media_query_width_and_vertical_scrollbars/

**Comment by squidfunk** [comment]
**TL;DR: fixed in #559, however TOC position not 100% correct**

The horizontal scrollbar should be fixed with #559, but this is really only a partial fix. The position of the right sidebar is not correct, which is because **some** browsers consider media query width to be the browser window width including the scrollbars, but the browser considers the browser window width without scrollbars when rendering the CSS. The breakpoint breaks to use `margin-left: 122rem` when the window is not really ready and cannot contain the element. All other alignment is not possible due to the fact what parent element can be set to overflow and which can be position relative. This is a really complicated issue. A possible solution could be to fix it with JavaScript, which is pretty ugly, but it's the easiest way.
